### PR TITLE
Update currentPercent for non-releasables to 0

### PR DESCRIPTION
**Pull request by dokipen**
Hello @ddbenson, @dnelson, @eblume, @silverlyra, @matkam, @micahnoland, @tonymeng, 

Please review the following commits I made in branch 'dokipen/20190916145720-0-post-canary':

- **Update currentPercent for non-releasables to 0** (af032dcfe6d8891468d312e3562e6a46d4cd95f7)

R=@ddbenson
R=@dnelson
R=@eblume
R=@silverlyra
R=@matkam
R=@micahnoland
R=@tonymeng

### CODE REVIEW

File: pkg/controller/releasemanager/incarnation.go [hunk 1]
```diff
@@ -566,6 +566,25 @@ func (i *IncarnationCollection) releasable() []Incarnation {
...
+	for _, incarnation := range i.releasable() {
+		releasableTags = append(releasableTags, incarnation.tag)
+	}
+
+	unreleasable := []Incarnation{}
+Outer:
+	for _, incarnation := range i.sorted() {
+		for _, tag := range releasableTags {
+			if incarnation.tag == tag {
+				continue Outer
```
Comment (silverlyra) [inline pkg/controller/releasemanager/incarnation.go]: fyi I own `ihatego.com`

File: pkg/controller/releasemanager/syncer.go [hunk 2]
```diff
@@ -286,6 +286,10 @@ func (r *ResourceSyncer) prepareRevisionsAndRules() ([]rmplan.Revision, []monito
 		}
 	}
 
+	for _, incarnation := range r.incarnations.unreleasable() {
```
Comment (silverlyra) [inline pkg/controller/releasemanager/syncer.go]: 💬can we add a comment here? what does this fix/ensure?

File: pkg/controller/releasemanager/incarnation.go (hunk 1 above)
Comment (dokipen) [reply pkg/controller/releasemanager/incarnation.go]: what's the big O of this function? 😂

Comment (dokipen) [reply pkg/controller/releasemanager/incarnation.go]: golang is trying to tell me to refactor using a map.

Comment (silverlyra) [reply pkg/controller/releasemanager/incarnation.go]: _tfw you steal many great ideas from Python – but forget the `set` type_ 😐

Comment (silverlyra) [reply pkg/controller/releasemanager/incarnation.go]: _we still wanted to capture some of the fundamental experience of writing C – misery_

File: pkg/controller/releasemanager/incarnation.go [hunk 3]
```diff
@@ -566,6 +566,25 @@ func (i *IncarnationCollection) releasable() []Incarnation {
...
 }
 
+func (i *IncarnationCollection) unreleasable() []Incarnation {
+	releasableTags := []string{}
+	for _, incarnation := range i.releasable() {
+		releasableTags = append(releasableTags, incarnation.tag)
+	}
+
+	unreleasable := []Incarnation{}
+Outer:
```
Comment (micahnoland) [inline pkg/controller/releasemanager/incarnation.go]: 😻 learned about label breaks, nice

Comment (dokipen) [reply pkg/controller/releasemanager/incarnation.go]: usually means that you're doing somethign wrong

**Pull request merged**